sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -276,7 +276,7 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       if (!tableExists) {
         val schema = JdbcUtils.schemaString(df, url)
         val sql = s"CREATE TABLE $table ($schema)"
-        conn.prepareStatement(sql).executeUpdate()
+        conn.createStatement.executeUpdate(sql)
```

Comment:
Hm, it occurs to me that there are several places where `prepareStatement` or `createStatement` are called, but the `Statement` is not explicitly `close()`ed. They're the two here that you are changing, plus a few others.

`JDBCRDD`:
```
      val rs = conn.prepareStatement(s"SELECT * FROM $table WHERE 1=0").executeQuery()
```
`JdbcUtils.tableExists`
```
    Try(conn.prepareStatement(dialect.getTableExistsQuery(table)).executeQuery()).isSuccess
```

I'm OK to commit this as-is since it's a fix for DDL, but you'd be welcome to also correct handling of the `Statement` here, and then while at it, correct the other 2 occurrences in the code base nearby.